client/my-sites/plans-grid/components/storage-add-on-dropdown.tsx

```diff
@@ -52,57 +67,79 @@ export const StorageAddOnDropdown = ( {
-		( select ) => {
-			return select( WpcomPlansUI.store ).getSelectedStorageOptionForPlan( planSlug );
-		},
+		( select ) => select( WpcomPlansUI.store ).getSelectedStorageOptionForPlan( planSlug ),
 		[ planSlug ]
 	);
-
-	// TODO: Consider transforming storageOptions outside of this component
 	const selectControlOptions = storageOptions.map( ( storageOption ) => {
 		const title = getStorageStringFromFeature( storageOption.slug ) || '';
 		const price = getStorageOptionPrice( storageAddOnsForPlan, storageOption.slug );
 		return {
-			key: storageOption?.slug,
-			name: <StorageAddOnOption title={ title } price={ price } showPrice={ showPrice } />,
+			key: storageOption.slug,
+			name: <StorageAddOnOption title={ title } price={ price } />,
 		};
 	} );
 
-	const defaultStorageOption = storageOptions.find( ( storageOption ) => ! storageOption?.isAddOn );
-	const selectedOptionKey = selectedStorageOptionForPlan || defaultStorageOption?.slug || '';
-	const selectedOptionPrice = getStorageOptionPrice( storageAddOnsForPlan, selectedOptionKey );
-	const selectedOptionTitle = getStorageStringFromFeature( selectedOptionKey ) || '';
+	const selectedOptionKey = selectedStorageOptionForPlan
+		? selectedStorageOptionForPlan
+		: storageOptions.find( ( storageOption ) => ! storageOption.isAddOn )?.slug;
+	const selectedOptionPrice =
+		selectedOptionKey && getStorageOptionPrice( storageAddOnsForPlan, selectedOptionKey );
+	const selectedOptionTitle =
+		( selectedOptionKey && getStorageStringFromFeature( selectedOptionKey ) ) || '';
+
 	const selectedOption = {
 		key: selectedOptionKey,
 		name: (
 			<StorageAddOnOption
 				title={ selectedOptionTitle }
 				price={ selectedOptionPrice }
-				showPrice={ showPrice }
+				isLargeCurrency={ isLargeCurrency }
+				priceOnSeparateLine={ priceOnSeparateLine }
 			/>
 		),
 	};
 	return (
-		<CustomSelectControl
-			label={ label }
-			options={ selectControlOptions }
-			value={ selectedOption }
-			onChange={ ( { selectedItem }: { selectedItem: { key: WPComStorageAddOnSlug } } ) => {
-				const addOnSlug = selectedItem?.key;
+		<>
+			<CustomSelectControl
+				label={ label }
+				options={ selectControlOptions }
+				value={ selectedOption }
+				onChange={ ( { selectedItem }: { selectedItem: { key: WPComStorageAddOnSlug } } ) => {
+					const addOnSlug = selectedItem?.key;
 
-				if ( addOnSlug ) {
-					onStorageAddOnClick && onStorageAddOnClick( addOnSlug );
-					setSelectedStorageOptionForPlan( { addOnSlug, planSlug } );
-				}
-			} }
-		/>
+					if ( addOnSlug ) {
+						onStorageAddOnClick && onStorageAddOnClick( addOnSlug );
+						setSelectedStorageOptionForPlan( { addOnSlug, planSlug } );
+					}
+				} }
+			/>
+			{ selectedOptionPrice && ( isLargeCurrency || priceOnSeparateLine ) && (
+				<div className="storage-add-on-dropdown__offset-price-container">
+					<span className="storage-add-on-dropdown__offset-price">
+						{ ` + ${ selectedOptionPrice }/${ translate( 'month' ) }` }
```

Comment:
Great catch. According to our i18n best practices PCYsg-3hI-p2(see the content regarding concatenation) they should be placeholders to maximize the flexibiliy of localization. However, let's fix this as a follow-up since I don't think it would be worth the wait of another week for translation if it doesn't sound like a terrible idea for you both :)

Reply:
Makes sense. We can definitely address this in a follow-up.

I think also holds for any of my other comments if they are not quick fixes. Personally, I'd be happy to review more `Plans 2023: Grid perf - ...` PRs :-D

^ up to you @jeyip

Reply:
Thanks for bringing this up @chriskmnds! And thanks for sharing the best practices doc @southp. I read through the docs and the concern with concatenation makes total sense 👍

I'll draft an issue and handle the i18n best practices in a higher priority follow up 😎